src/errors.ts

```diff
@@ -1,5 +1,31 @@
 import * as z from 'zod'
-import type { ErrorWithMessage, SchemaError } from './types'
+import type { ErrorData, ErrorWithMessage, SchemaError } from './types'
+
+function parseError(error: unknown): ErrorData {
```

Comment:
I would rather have narrower types for potential errors since programmers are mostly in control. this would eliminate the need for a `parseError` and simplify the PR  without compromising the DX.